### PR TITLE
Add grafana to monitoring

### PR DESCRIPTION
**Pull request by idoqo**
This adds a new grafana container as part of #73. It also fixes the issue with empty dockerHostAddr (which happens when the container is in a "stopped" state and has no IP address or gateway).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,20 @@ export CLIENT_SECRET=<GITHUB_CLIENT_SECRET>
...
+for you on startup. Every new database you add will automatically be added to postgres exporter for scraping and its metrics exposed in Prometheus/Grafana.
 
+To enable monitoring, set the `common.monitoring` field in your spinup config file to `true` (or add it if it doesn't exist).
+*Note that `common.monitoring` above implies the `monitoring` field in the `common` section in the YAML file*
+When monitoring is enabled, the monitoring services are started as follows:
+- Prometheus - http://localhost:9090
+- Grafana - http://localhost:9091
+- Postgres Exporter - http://localhost:9187
+
+Visit http://localhost:9091/explore o explore the provisioned Prometheus in Grafana,
```
Comment (viggy28) [inline README.md:48]: I think a typo `to`.

File: internal/monitor/monitor.go [hunk 2]
```diff
@@ -46,6 +47,11 @@ func (r *Runtime) BootstrapServices(ctx context.Context) error {
 				stopErr := pgExporterContainer.Stop(ctx, r.dockerClient, types.ContainerStartOptions{})
 				r.logger.Error("stopping exporter container", zap.Error(stopErr))
 			}
+
+			if gfContainer != nil {
+				stopErr := gfContainer.Stop(ctx, r.dockerClient, types.ContainerStartOptions{})
+				r.logger.Error("stopping grafana container", zap.Error(stopErr))
```
Comment (viggy28) [inline internal/monitor/monitor.go:53]: Do we always want to print the `stopErr` value? (even when the error is nil)

File: internal/monitor/monitor.go [hunk 3]
```diff
@@ -215,3 +286,34 @@ func (r *Runtime) writePromConfig(cfgPath string) error {
...
+
+	return cfgPath, err
+}
+
+func (r *Runtime) writeGrafanaConfig(cfgPath string) error {
+	cfg := fmt.Sprintf(`# prometheus provisioning file for grafana
+apiVersion: 1
+
+datasources:
+- name: Prometheus'
```
Comment (viggy28) [inline internal/monitor/monitor.go:305]: Is the backtick needed here or its missing one more?

**Review by viggy28 — APPROVED** [review]

File: internal/monitor/monitor.go (hunk 2 above)
Comment (idoqo) [reply internal/monitor/monitor.go:53]: Right, will clean it up

**Review by skant7 — APPROVED** [review]

**Review by viggy28 — APPROVED** [review]
LGTM

**Pull request merged**